metecho/api/gh.py

```diff
@@ -35,7 +38,77 @@ class NoGitHubTokenError(Exception):
     pass
 
 
-def gh_given_user(user):
+def copy_branch_protection(source: Branch, target: Branch):
+    """
+    Copy the branch protection [output][1] of one branch into the [input][2] of another.
+
+    As of May 2022 it appears the following settings are only available to the GUI, not
+    the API:
+
+    - Require signed commits
+    - Require deployments to succeed before merging
+    - Restrict pushes that create matching branches
```

Comment:
@jerivas When I create new repos using https://github.com/jerivastest/cci-blueprint/ as the template, this box is automatically checked after creation.

https://github.com/jerivastest/test-branch-protection/settings/branch_protection_rules/25966244

![Screen Shot 2022-05-25 at 10 10 25 AM](https://user-images.githubusercontent.com/552316/170282640-ef0977cf-3ce7-4b70-853a-5dd96374834c.jpg)

Reply:
Huh, interesting. I could never get it to copy on my previous tests, but it makes sense that it is supported because the parent checkbox is supported and documented. I'll test again and remove it from the list

Reply:
Fixed in 2375dd6cf586da02a6657f8c35aa286c988e06cc